src/Peripherals/PeripheralInterface.h

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+class QString;
+
+class PeripheralInterface
+{
+public:
+    virtual bool attemptConnection()=0;
+    virtual void setPortName(QString portname)=0;
+    virtual void sendData(const unsigned char* data, int length)=0;
```

Comment:
the `=` should have spaces around it. We should also add a virtual destructor here.
http://stackoverflow.com/questions/3628529/should-c-interfaces-have-a-virtual-destructor